src/sage/groups/libgap_mixin.py

```diff
@@ -963,3 +965,128 @@ def is_isomorphic(self, H):
+            C library interface to GAP
+    """
+    if not isinstance(group, GapElement):
+        try:
+            group = group._libgap_()
+        except AttributeError:
+            raise NotImplementedError("only implemented for groups that contain a working __libgap__ mehtod")
+
+    if not group.IsFinite().sage():
+        raise NotImplementedError("only implemented for finite groups")
+
+    # A function to check if the group is generated by the given generators or not.
+    def is_groupbygens(group, gens):
+        if not isinstance(group, GapElement):
+            group = group._libgap_()
+        return set(group.AsList()) == set(libgap.GroupByGenerators(gens).AsList())
+
+    group_elements = group.AsList()
+
+    if group.IsCyclic().sage():
+        for ele in group_elements:
+            if is_groupbygens(group, [ele]):
+                return [ele]
+
+    if group.IsSimple().sage():
+        n = len(group_elements)
+        for i in range(n):
+            for j in range(i+1, n):
+                if is_groupbygens(group, [group_elements[i], group_elements[j]]):
+                    return [group_elements[i], group_elements[j]]
+
+    N = group.MinimalNormalSubgroups()[0]
+    n = N.SmallGeneratingSet()
+
+    phi = group.NaturalHomomorphismByNormalSubgroup(N)
+    GbyN = phi.ImagesSource()
+    GbyN_mingenset = minimum_generating_set(GbyN)
+
+    g = [phi.PreImagesRepresentative(g) for g in GbyN_mingenset]
+    l = len(g)
+
+    if N.IsAbelian().sage():
+        if is_groupbygens(group, g):
+            return g
+
+        for i in range(l):
+            for j in range(len(n)):
+                temp = g[i]
+                g[i] = g[i]*n[j]
+                if is_groupbygens(group, g):
+                    return g
+                g[i] = temp
+
+        return g + [n[0]]
+
+    # A function to generate some combinations of the generators
+    # of the group according to the algorithm.
+    def gen_combinations(g, N, t):
+        if t > len(g):
+            t = len(g)
+        if t <= 0:
+            yield g
+        for go in gen_combinations(g,N,t-1):
+            for j in range(len(N)):
+                temp = go[t-1]
+                go[t-1] = go[t-1]*N[j]
+                yield go
+                go[t-1] = temp
+
+    # Here, the value of t is different from the one in the research paper mentioned
+    # in the Algorithm section becuase in the paper the value of t was calculated using
+    # cheif series of the group but currently sage does not have cheif series implemented.
+    t = ceil((13/5 + log(group.Size().sage(), 2) / log(N.Size().sage(), 2)))
+    N_list = N.AsList()
+
+    for nl in N_list:
+        for raw_gens in gen_combinations(g, N_list, t):
+            if nl == N_list[0]:   # considering N_list[0] is the identity element
+                gens = raw_gens
+            else:
+                gens = raw_gens+[nl]
+
+            if is_groupbygens(group, gens):
+                return gens
```

Comment:
What happens if it happens to reach the end of the function? I think you should have an `assert` for that case (that should never happen, right?).

Reply:
Yes, that will never happen according to the algorithm.

Reply:
You should add a message error saying that it should not fall into here.